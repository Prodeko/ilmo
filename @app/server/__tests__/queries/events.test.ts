import {
  createEventDataAndLogin,
  createUserAndLogIn,
  deleteTestData,
  runGraphQLQuery,
  setup,
  teardown,
} from "../helpers"

beforeEach(deleteTestData)
beforeAll(setup)
afterAll(teardown)

const eventsQuery = `
{
  events {
    nodes {
      id
    }
  }
}`

test("events when logged out (RLS policy)", async () => {
  await createEventDataAndLogin({
    eventOptions: { create: true, amount: 5, isDraft: true },
  })

  await runGraphQLQuery(
    // GraphQL query goes here:
    eventsQuery,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
    {},

    // This function runs all your test assertions:
    async (json) => {
      expect(json.errors).toBeFalsy()
      expect(json.data).toBeTruthy()
      expect(json.data!.events.nodes).toHaveLength(0)
    }
  )
})

test("events when logged in, NOT admin and NOT member of the owner organization (RLS policy)", async () => {
  await createEventDataAndLogin({
    eventOptions: { create: true, amount: 5, isDraft: true },
  })

  // Different user and session than the one who created the events.
  // The user is NOT admin so draft events should NOT be visible to them.
  const { session } = await createUserAndLogIn()

  await runGraphQLQuery(
    // GraphQL query goes here:
    eventsQuery,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
    {
      user: { sessionId: session.uuid },
    },

    // This function runs all your test assertions:
    async (json) => {
      expect(json.errors).toBeFalsy()
      expect(json.data).toBeTruthy()
      expect(json.data!.events.nodes).toHaveLength(0)
    }
  )
})

test("events when logged in, NOT admin and IS member of the owner organization (RLS policy)", async () => {
  const { session } = await createEventDataAndLogin({
    eventOptions: { create: true, amount: 5, isDraft: true },
  })

  await runGraphQLQuery(
    // GraphQL query goes here:
    eventsQuery,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
    {
      user: { sessionId: session.uuid },
    },

    // This function runs all your test assertions:
    async (json) => {
      expect(json.errors).toBeFalsy()
      expect(json.data).toBeTruthy()
      expect(json.data!.events.nodes).toHaveLength(5)
    }
  )
})

test("events when logged in, IS admin, NOT a member of the owner organization (RLS policy)", async () => {
  await createEventDataAndLogin({
    eventOptions: { create: true, amount: 5, isDraft: true },
  })

  // Different user and session than the one who created the events.
  // The user is admin so draft events should be visible to them.
  const { session } = await createUserAndLogIn({ isAdmin: true })

  await runGraphQLQuery(
    // GraphQL query goes here:
    eventsQuery,

    // GraphQL variables:
    {},

    // Additional props to add to `req` (e.g. `user: {sessionId: '...'}`)
    {
      user: { sessionId: session.uuid },
    },

    // This function runs all your test assertions:
    async (json) => {
      expect(json.errors).toBeFalsy()
      expect(json.data).toBeTruthy()
      expect(json.data!.events.nodes).toHaveLength(5)
    }
  )
})
