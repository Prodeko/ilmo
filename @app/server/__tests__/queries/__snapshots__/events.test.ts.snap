// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events when logged in, IS admin, NOT a member of the owner organization (RLS policy) 1`] = `
Object {
  "data": Object {
    "events": Object {
      "nodes": Array [
        Object {
          "id": "[id-1]",
        },
        Object {
          "id": "[id-2]",
        },
        Object {
          "id": "[id-3]",
        },
        Object {
          "id": "[id-4]",
        },
        Object {
          "id": "[id-5]",
        },
      ],
    },
  },
}
`;

exports[`events when logged in, NOT admin and IS member of the owner organization (RLS policy) 1`] = `
Object {
  "data": Object {
    "events": Object {
      "nodes": Array [
        Object {
          "id": "[id-1]",
        },
        Object {
          "id": "[id-2]",
        },
        Object {
          "id": "[id-3]",
        },
        Object {
          "id": "[id-4]",
        },
        Object {
          "id": "[id-5]",
        },
      ],
    },
  },
}
`;

exports[`events when logged in, NOT admin and NOT member of the owner organization (RLS policy) 1`] = `
{
  "data": {
    "events": {
      "nodes": [],
    },
  },
}
`;

exports[`events when logged out (RLS policy) 1`] = `
{
  "data": {
    "events": {
      "nodes": [],
    },
  },
}
`;
